skbio/alignment/_tabular_msa.py

```diff
@@ -685,13 +674,14 @@ def reindex(self, key=None, keys=None):
         array(['a', 'b'], dtype=object)
 
         """
-        if key is not None and keys is not None:
+        if minter is not None and keys is not None:
             raise ValueError(
-                "Cannot use both `key` and `keys` at the same time.")
+                "Cannot use both `minter` and `keys` at the same time.")
 
         keys_ = None
-        if key is not None:
-            keys_ = [resolve_key(seq, key) for seq in self._seqs]
+        if minter is not None:
+            self._minter = minter
```

Comment:
Can you refactor to avoid modifying object state until after we're sure no errors will be thrown? The line below could fail if the user provides an invalid minter.